In [35]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
from sklearn import linear_model

In [36]:
#Multiple regression enables the prediction to be more accurate as it uses more variables
df = pd.read_csv(r"C:\Users\sanke\Downloads\data.csv")
df.head(5)

,Car,Model,Volume,Weight,CO2
0,Toyoty,Aygo,1000,790,99
1,Mitsubishi,Space Star,1200,1160,95
2,Skoda,Citigo,1000,929,95
3,Fiat,500,900,865,90
4,Mini,Cooper,1500,1140,105


## Determining how much CO2 a car emmits based on weight and volume

In [37]:
#multiples regression uses more x variables 
x = df[['Weight','Volume']]
y = df['CO2']

In [38]:
#creating a linear regression object
regr= linear_model.LinearRegression()
regr.fit(x,y)
print(regr.coef_)

[0.00755095 0.00780526]


In [39]:
#predicting the  co2 emission of a car of weight = 2300 kg and volume = 1300cm cubed
predictedC02 = regr.predict([[2300,1300]])
print("A car with 2300 kgs and a volume of 1300 cm cubed releases approximately :" , predictedC02)

A car with 2300 kgs and a volume of 1300 cm cubed releases approximately : [107.2087328]


c:\Users\sanke\anaconda3\Lib\site-packages\sklearn\base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [41]:
#using a dataset from seaborn
df_taxis = sns.load_dataset('taxis')
df_taxis

,pickup,dropoff,passengers,distance,fare,tip,tolls,total,color,payment,pickup_zone,dropoff_zone,pickup_borough,dropoff_borough
0,2019-03-23 20:21:09,2019-03-23 20:27:24,1,1.60,7.0,2.15,0.0,12.95,yellow,credit card,Lenox Hill West,UN/Turtle Bay South,Manhattan,Manhattan
1,2019-03-04 16:11:55,2019-03-04 16:19:00,1,0.79,5.0,0.00,0.0,9.30,yellow,cash,Upper West Side South,Upper West Side South,Manhattan,Manhattan
2,2019-03-27 17:53:01,2019-03-27 18:00:25,1,1.37,7.5,2.36,0.0,14.16,yellow,credit card,Alphabet City,West Village,Manhattan,Manhattan
3,2019-03-10 01:23:59,2019-03-10 01:49:51,1,7.70,27.0,6.15,0.0,36.95,yellow,credit card,Hudson Sq,Yorkville West,Manhattan,Manhattan
4,2019-03-30 13:27:42,2019-03-30 13:37:14,3,2.16,9.0,1.10,0.0,13.40,yellow,credit card,Midtown East,Yorkville West,Manhattan,Manhattan
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6428,2019-03-31 09:51:53,2019-03-31 09:55:27,1,0.75,4.5,1.06,0.0,6.36,green,credit card,East Harlem North,Central Harlem North,Manhattan,Manhattan
6429,2019-03-31 17:38:00,2019-03-31 18:34:23,1,18.74,58.0,0.00,0.0,58.80,green,credit card,Jamaica,East Concourse/Concourse Village,Queens,Bronx
6430,2019-03-23 22:55:18,2019-03-23 23:14:25,1,4.14,16.0,0.00,0.0,17.30,green,cash,Crown Heights North,Bushwick North,Brooklyn,Brooklyn
6431,2019-03-04 10:09:25,2019-03-04 10:14:29,1,1.12,6.0,0.00,0.0,6.80,green,credit card,East New York,East Flatbush/Remsen Village,Brooklyn,Brooklyn


- Determining approximately how much tip a driver gets based on fare and distance travelled

In [43]:
from sklearn import linear_model
a = df_taxis[['distance','fare']]
b = df_taxis['tip']

In [47]:
regr = linear_model.LinearRegression()
regr.fit(a,b)

LinearRegression()

In [48]:
print(regr.coef_)

[0.01256227 0.09973781]


- The coefficient 0.01256227 tells us that if distance increased by 1 km then the tip offered increases by 0.01256227

- The coefficient 0.09973781 : if fare increses by one dollar then the tip offered increases by 0.09973781

In [52]:
predicted_tip = regr.predict([[5,25]])
print('The tip offered for a distance of 5 km and fare of $25 is', predicted_tip)

The tip offered for a distance of 5 km and fare of $25 is [3.19180526]


c:\Users\sanke\anaconda3\Lib\site-packages\sklearn\base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
